storage/mysql/admin_storage.go

```diff
@@ -320,6 +331,11 @@ func (t *adminTX) CreateTree(ctx context.Context, tree *trillian.Tree) (*trillia
 		return nil, fmt.Errorf("unexpected DuplicatePolicy value: %v", newTree.DuplicatePolicy)
 	}
 
+	privateKey, err := proto.Marshal(newTree.PrivateKey)
```

Comment:
Should we validate private key types we know of before marshaling? For example, for PEMKeyFile, we could check that the file exists, password works, so on.

There's some generic validation on storage/tree_validation.go. We should at least verify that the Any's TypeUrl and Value are not empty there, I think.